examples/plot_bcic_iv_4_ecog_cropped.py

```diff
@@ -0,0 +1,381 @@
+#
+
+import torch
+from braindecode.util import set_random_seeds
+from braindecode.models import ShallowFBCSPNet
+
+cuda = torch.cuda.is_available()  # check if GPU is available, if True chooses to use it
+device = 'cuda' if cuda else 'cpu'
+if cuda:
+    torch.backends.cudnn.benchmark = True
+seed = 20200220  # random seed to make results reproducible
+# Set random seed to be able to reproduce results
+set_random_seeds(seed=seed, cuda=cuda)
+
+n_classes = 5
+# Extract number of chans and time steps from dataset
+n_chans = 62
+
+model = ShallowFBCSPNet(
+    n_chans,
+    n_classes,
+    final_conv_length=2,
+)
+# We are removing the softmax layer to make it a regression model
+new_model = torch.nn.Sequential()
+for name, module_ in model.named_children():
+    if "softmax" in name:
+        continue
+    new_model.add_module(name, module_)
+model = new_model
+
+# Send model to GPU
+if cuda:
+    model.cuda()
+
+from braindecode.models.util import to_dense_prediction_model, get_output_shape
+to_dense_prediction_model(model)
+
+
+######################################################################
+# To know the models’ receptive field, we calculate the shape of model
+# output for a dummy input.
+
+n_preds_per_input = get_output_shape(model, n_chans, input_window_samples)[2]
+
+
+######################################################################
+# Cut Compute Windows
+# ~~~~~~~~~~~~~~~~~~~
+#
+
+
+from braindecode.preprocessing.windowers import create_fixed_length_windows
+
+# Create windows using braindecode function for this. It needs parameters to define how
+# trials should be used.
+
+windows_dataset = create_fixed_length_windows(
+    dataset,
+    start_offset_samples=0,
+    stop_offset_samples=None,
+    window_size_samples=input_window_samples,
+    window_stride_samples=n_preds_per_input,
+    drop_last_window=False,
+    targets_from='channels',
+    last_target_only=False,
+    preload=False
+)
+
+
+######################################################################
+# Split dataset into train and valid
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+######################################################################
+# We can easily split the dataset using additional info stored in the
+# description attribute, in this case ``session`` column.
+
+subsets = windows_dataset.split('session')
+train_set = subsets['train']
+test_set = subsets['test']
+
+import torch
+from sklearn.model_selection import train_test_split
+
+# We can split train dataset into training and validation datasets using
+# `sklearn.model_selection.train_test_split` and `torch.utils.data.Subset`
+idx_train, idx_valid = train_test_split(np.arange(len(train_set)),
+                                        random_state=100,
+                                        test_size=0.2,
+                                        shuffle=False)
+valid_set = test_set
+# valid_set = torch.utils.data.Subset(train_set, idx_valid)
```

Comment:
@robintibor is there any way to subset a cropped dataset without raising error while validating that a windows indices have to start from zero?

Reply:
hmhm very good question. I guess no and we maybe should implement it? I think usually our expectation was that splitting is mostly done before windowing. However, I have also split valid datasets off windows/cropped datasets, but there usually I knew how many windows there were per trial and could ensure by hand that everything will be correct.. probably we should have some function for it? But to not delay this PR further is it possible to move the splitting to before the windowing?